gimme

```diff
@@ -81,130 +59,175 @@ _source() {
 		"https://go.googlecode.com/files/go${1}.src.tar.gz"
 		"https://go.googlecode.com/files/go.${1}.src.tar.gz"
 	)
-	_do_curls "$2" "${urls[@]}"
+	_do_curls "${2}" "${urls[@]}"
 }
 
 # _fetch "dir"
 _fetch() {
-	mkdir -p "$(dirname "$1")"
-	
-	if [ -d "$1/.git" ]; then
+	mkdir -p "$(dirname "${1}")"
+
+	if [[ -d "${1}/.git" ]] ; then
 		(
-			cd "$1"
-			git remote set-url origin "$GIMME_GO_GIT_REMOTE"
+			cd "${1}"
+			git remote set-url origin "${GIMME_GO_GIT_REMOTE}"
 			git fetch -q --all && git fetch -q --tags
 		)
 		return
 	fi
-	
-	git clone -q "$GIMME_GO_GIT_REMOTE" "$1"
+
+	git clone -q "${GIMME_GO_GIT_REMOTE}" "${1}"
 }
 
 # _checkout "version" "dir"
 _checkout() {
-	_fetch "$2"
-	( cd "$2" && {
-		git reset -q --hard "origin/$1" \
-		|| git reset -q --hard "origin/go$1" \
-		|| { [ "$1" = 'tip' ] && git reset -q --hard origin/master; } \
-		|| git reset -q --hard "refs/tags/$1" \
-		|| git reset -q --hard "refs/tags/go$1"
+	_fetch "${2}"
+	( cd "${2}" && {
+		git reset -q --hard "origin/${1}" \
+		|| git reset -q --hard "origin/go${1}" \
+		|| { [ "${1}" = 'tip' ] && git reset -q --hard origin/master ; } \
+		|| git reset -q --hard "refs/tags/${1}" \
+		|| git reset -q --hard "refs/tags/go${1}"
 	} 2>/dev/null )
 }
 
 # _extract "file.tar.gz" "dir"
 _extract() {
-	mkdir -p "$2"
-	tar -xf "$1" -C "$2" --strip-components 1
+	mkdir -p "${2}"
+	tar -xf "${1}" -C "${2}" --strip-components 1
 }
 
 # _compile "dir"
 _compile() {
 	(
-		cd "$1"
-		if [ -d .git ]; then
+		local bootstrap_root="${GIMME_VERSION_PREFIX}/go1.4.${GIMME_OS}.${GIMME_ARCH}"
+		if [ ! -d "${bootstrap_root}" ] ; then
+			echo >&2 "error: Missing bootstrap installation at ${bootstrap_root}"
+			echo >&2 "error: Unable to compile go from git clone"
+			return
```

Comment:
I'm not entirely sure what this bit is for -- can you elaborate?  I don't think I've ever run into the issue this is solving, so I want to make sure I understand it (because I'm sure to run into it later otherwise :angel:).